packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -89,10 +128,28 @@ export function generateForm(
         `;
     }
 
-    if (addMode && createMutation) {
+    if (addMode && createMutationType) {
         gqlDocuments[`create${gqlType}Mutation`] = `
-            mutation Create${gqlType}($input: ${gqlType}Input!) {
-                ${createMutation.name}(input: $input) {
+            mutation Create${gqlType}(${
+            gqlArgs.filter((gqlArg) => !gqlArg.isInputArgSubfield && gqlArg.queryOrMutationName === `create${gqlType}`).length
+                ? `${gqlArgs
+                      .filter((gqlArg) => !gqlArg.isInputArgSubfield && gqlArg.queryOrMutationName === `create${gqlType}`)
+                      .map((gqlArg) => {
+                          return `$${gqlArg.name}: ${gqlArg.type}!`;
+                      })
+                      .join(", ")}, `
+                : ``
```

Comment:
You don't need this length checks here, since joining an empty string array results in an empty string anyway

Reply:
it's about this unremarkable comma after join. .join(", ")}**,** ` this would be too much... honestly I'm no fan of this string-concatenation stuff. I would prefer a function generating this mutation-params string hiding away 20 lines of code... but this will hopefully part of a later refactoring :-)